extensions/script-libraries/semver/semver.livecodescript

```diff
@@ -0,0 +1,726 @@
+   if the number of items of tVersion > 1 then
+      put the last item of tVersion into tParsedVersion["metadata"]
+      delete the last item of tVersion
+   end if
+   
+   -- prerelease stage if any -> major.minor.patch-prerelease.build
+   set the itemDelimiter to "-"
+   if the number of items of tVersion > 1 then
+      local tPrerelease
+      put item 2 to -1 of tVersion into tPrerelease
+      delete item 2 to -1 of tVersion
+      
+      set the itemDelimiter to "."
+      put item 1 of tPrerelease into tParsedVersion["prerelease"]
+      put item 2 of tPrerelease into tParsedVersion["build"]
+      if tParsedVersion["build"] is empty then
+         put 0 into tParsedVersion["build"]
+      end if
+   end if
+   
+   
+   -- major number -> major.minor.patch
+   set the itemDelimiter to "."
+   put item 1 of tVersion into tParsedVersion["major"]
+   if tParsedVersion["major"] is empty then
+      if pRange then
+         put "x" into tParsedVersion["major"]
+      else
+         put 0 into tParsedVersion["major"]
+      end if
+   end if
+   if pRange then
+      replace "*" with "x" in tParsedVersion["major"]
+   end if
+   delete item 1 of tVersion
+   
+   -- minor number -> minor.patch
+   put item 1 of tVersion into tParsedVersion["minor"]
+   if tParsedVersion["minor"] is empty then
+      if pRange then
+         put "x" into tParsedVersion["minor"]
+      else
+         put 0 into tParsedVersion["minor"]
+      end if
+   end if
+   if pRange then
+      replace "*" with "x" in tParsedVersion["minor"]
+   end if
+   delete item 1 of tVersion
+   
+   -- patch number -> patch
+   put tVersion into tParsedVersion["patch"]
+   if tParsedVersion["patch"] is empty then
+      if pRange then
+         put "x" into tParsedVersion["patch"]
+      else
+         put 0 into tParsedVersion["patch"]
+      end if
+   end if
+   if pRange then
+      replace "*" with "x" in tParsedVersion["patch"]
+   end if
+   
+   if tParsedVersion["major"] is not an integer \
+         and not (pRange and tParsedVersion["major"] is "x") then
+      return "Invalid major number:" && xVersion for error
+   end if
+   
+   if tParsedVersion["minor"] is not an integer \
+         and not (pRange and tParsedVersion["minor"] is "x") then
+      return "Invalid minor number:" && xVersion for error
+   end if
+   
+   if tParsedVersion["patch"] is not an integer \
+         and not (pRange and tParsedVersion["patch"] is "x") then
+      return "Invalid patch number:" && xVersion for error
+   end if
+   
+   if tParsedVersion["prerelease"] is not empty and tParsedVersion["build"] is not an integer then
+      return "Invalid build number string:" && xVersion for error
+   end if
+   
+   put tParsedVersion into xVersion
+end __SemverParse
+
+---------------------------------------------------
+-- COMPARISON
+
+private function __GreaterThan pLeft, pRight
+   if pLeft["major"] is "x" or pRight["major"] is "x" then
+      return true
+   end if
```

Comment:
Yes it probably does... I think this bit probably applies even though it's not discussing X-Ranges:
```
- ~1.2.3-beta.2 := >=1.2.3-beta.2 <1.3.0 Note that prereleases in the 1.2.3 version will be
allowed, if they are greater than or equal to beta.2. So, 1.2.3-beta.4 would be allowed, but
`1.2.4-beta.2` would not, because it is a prerelease of a different [major, minor, patch] tuple.
```